.github/workflows/go.yml

```diff
@@ -0,0 +1,54 @@
+name: Go
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.13
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
+
+    - name: Build main
+      run: go build main.go
+
+    - name: Run Go Tests
+      run: go test ./...
+    
+    - name: Check format
```

Comment:
It's not obvious but as far as I see now - it's better to check go.sum first and them `go fmt` because `go fmt` could also add missing dependencies to go.sum. See https://github.com/devfile/parser/runs/905595388?check_suite_focus=true

Reply:
+1 I ran into that as well.